source/Popup/Views/Deposit/index.jsx

```diff
@@ -1,20 +1,95 @@
-import React from 'react';
-import { Layout } from '@components';
-import { Header } from '@ui';
-import useSteps from './hooks/useSteps';
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import { Typography } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import Grid from '@material-ui/core/Grid';
+import { useSelector } from 'react-redux';
+
+import { useRouter } from '@components/Router';
+import {
+  QRCode, CopyButton, IconQrCode, Layout,
+} from '@components';
+import shortAddress from '@shared/utils/short-address';
+import {
+  Header, LinkButton, InputBase, Dialog, Container,
+} from '@ui';
+import useStyles from './styles';
 
 const Deposit = () => {
-  const {
-    component,
-    left,
-    right,
-    center,
-  } = useSteps();
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { navigator } = useRouter();
+  const [openQr, setOpenQr] = useState(false);
+  const { principalId, accountId } = useSelector((state) => state.wallet);
 
   return (
     <Layout>
-      <Header left={left} center={center} right={right} />
-      {component}
+      <Header
+        left={null}
+        center={t('deposit.title')}
+        right={
+          <LinkButton value={t('common.done')} onClick={() => navigator.navigate('home')} />
+        }
+      />
+      <Container>
+        <Grid container spacing={1}>
+          <Grid item xs={12}>
+            <Typography variant="h5">{t('deposit.depositIcpTitle1')}</Typography>
+          </Grid>
+          <Grid item xs={12}>
+            <Typography variant="subtitle2">{t('deposit.depositIcpSubtitle1')}</Typography>
+          </Grid>
+          <Grid item xs={12}>
+            <InputBase>
+              <div className={classes.addressContainer}>
+                <Typography variant="h4" style={{ marginRight: 'auto' }}>{shortAddress(principalId)}</Typography>
```

Comment:
I say we pass in custom values to the lengths to make it look a bit better here and below.